src/pkg/packager/sources/sources.go

```diff
@@ -41,51 +41,64 @@ type PackageSource interface {
 }
 
 // Identify returns the type of package source based on the provided package source string.
-func Identify(pkgSrc string) string {
-	if helpers.IsURL(pkgSrc) {
-		parsed, _ := url.Parse(pkgSrc)
+func Identify(src string) string {
+	if helpers.IsURL(src) {
+		parsed, _ := url.Parse(src)
 		return parsed.Scheme
 	}
 
-	if strings.Contains(pkgSrc, ".part000") {
+	if strings.Contains(src, ".part000") {
 		return "split"
 	}
 
-	if IsValidFileExtension(pkgSrc) {
+	if IsValidFileExtension(src) {
 		return "tarball"
 	}
 
 	return ""
 }
 
 // New returns a new PackageSource based on the provided package options.
-func New(pkgOpts *types.ZarfPackageOptions) (PackageSource, error) {
+func New(src, shasum, publicKeyPath, sGetKeyPath string) (PackageSource, error) {
 	var source PackageSource
-
-	pkgSrc := pkgOpts.PackageSource
-
-	switch Identify(pkgSrc) {
+	switch Identify(src) {
 	case "oci":
-		if pkgOpts.Shasum != "" {
-			pkgSrc = fmt.Sprintf("%s@sha256:%s", pkgSrc, pkgOpts.Shasum)
+		ociSrc := src
```

Comment:
This seems like this is done for clarity but does not follow Go idioms - I would wager to remove.